pkg/datapath/linux/node_linux_test.go

```diff
@@ -1026,18 +1027,41 @@ func (s *linuxPrivilegedIPv4OnlyTestSuite) TestArpPingHandling(c *check.C) {
 	})
 	c.Assert(err, check.IsNil)
 
+	wait := func(now time.Time, nodeID nodeTypes.Identity, isDelete bool) {
+		err := testutils.WaitUntil(func() bool {
+			linuxNodeHandler.neighLock.Lock()
+			defer linuxNodeHandler.neighLock.Unlock()
+			nextHop, found := linuxNodeHandler.neighNextHopByNode[nodeID]
+			if !found {
+				return isDelete
+			}
+			lastPing, found := linuxNodeHandler.neighLastPingByNextHop[nextHop]
+			if !found {
+				return isDelete
+			}
+			if isDelete {
+				return false
+			}
+			return now.Before(lastPing)
+		}, 5*time.Second)
+		c.Assert(err, check.IsNil)
+	}
+
 	nodev1 := nodeTypes.Node{
 		Name:        "node1",
 		IPAddresses: []nodeTypes.Address{{nodeaddressing.NodeInternalIP, ip1}},
 	}
+	now := time.Now()
 	err = linuxNodeHandler.NodeAdd(nodev1)
 	c.Assert(err, check.IsNil)
+	// insertNeighbor is invoked async
```

Comment:
The same comment appears a few lines below. Was this maybe forgotten from an earlier version?

Reply:
Removed redundant comments.